packages/medusa/src/repositories/staged-job.ts

```diff
@@ -1,5 +1,37 @@
 import { EntityRepository, Repository } from "typeorm"
-import { StagedJob } from "../models/staged-job"
+import { StagedJob } from "../models"
+import { QueryDeepPartialEntity } from "typeorm/query-builder/QueryPartialEntity"
+import { RawSqlResultsToEntityTransformer } from "typeorm/query-builder/transformer/RawSqlResultsToEntityTransformer"
+import { RelationIdLoader } from "typeorm/query-builder/relation-id/RelationIdLoader"
 
 @EntityRepository(StagedJob)
-export class StagedJobRepository extends Repository<StagedJob> {}
+export class StagedJobRepository extends Repository<StagedJob> {
+  async insertBulk(jobToCreates: QueryDeepPartialEntity<StagedJob>[]) {
+    const queryBuilder = this.createQueryBuilder()
+    const rawStagedJobs = await queryBuilder
+      .insert()
+      .into(StagedJob)
+      .values(jobToCreates)
+      .returning("*")
+      .execute()
+
+    const relationIdLoader = new RelationIdLoader(
+      queryBuilder.connection,
+      this.queryRunner,
+      queryBuilder.expressionMap.relationIdAttributes
+    )
+    const rawRelationIdResults = await relationIdLoader.load(rawStagedJobs.raw)
+    const transformer = new RawSqlResultsToEntityTransformer(
+      queryBuilder.expressionMap,
+      queryBuilder.connection.driver,
+      rawRelationIdResults,
+      [],
+      this.queryRunner
+    )
+
+    return transformer.transform(
+      rawStagedJobs.raw,
+      queryBuilder.expressionMap.mainAlias!
+    ) as StagedJob[]
+  }
```

Comment:
thought(non-blocking): According to this [issue](https://github.com/typeorm/typeorm/issues/1663), there's a simpler way to transform rows into entities. 

I haven't tested it my self, but it looks like we can do:
```
    const result = (await queryBuilder
        .insert()
        .into(StagedJob)
        .values(jobToCreates)
        .returning('*')
        .execute()
    ).generatedMaps as StagedJob[];
```

If this is the case, I'd rather go with that, as the current implementation is quite hard to comprehend without digging into the Typeorm codebase.